code/modules/vehicles/tonk.dm

```diff
@@ -0,0 +1,354 @@
+	var/obj/turret_overlay/turret_overlay //Allows for independantly swivelling guns, wow!
+	var/obj/turret_overlay/minigun/minigun_overlay
+	var/max_health = 300 //i dunno what a good number for this is, change this maintainers!
+	var/obj/item/weapon/tank_weapon/main_cannon //What we use to shoot big shells
+	var/obj/item/weapon/tank_weapon/minigun/minigun //What we use to shoot mini shells ((rapidfire xenocrusher 6000))
+	var/main_cannon_dir = null //So that the guns swivel independantly
+	var/minigun_dir = null
+	var/firing = FALSE //Used in autofire
+	var/turf/stored_firing_turf = null
+
+/obj/turret_overlay
+	name = "Tank gun turret"
+	desc = "Big fucking tonk gun oh lawd he comin"
+	icon = 'icons/obj/tonk_gun.dmi'
+	icon_state = "turret"
+	layer = 3.1
+	animate_movement = TRUE //So it doesnt just ping back and forth and look all stupid
+	mouse_opacity = FALSE //It's an overlay
+
+/obj/vehicle/tonk/Initialize()
+	. = ..()
+	turret_overlay = new(get_turf(src))
+	update_icon()
+	health = max_health
+	main_cannon = new(src) //Make our guns
+	minigun = new(src)
+	main_cannon.owner = src
+	minigun.owner = src
+
+/obj/vehicle/tonk/Destroy()
+	qdel(turret_overlay)
+	qdel(main_cannon)
+	qdel(minigun)
+	. = ..()
+
+/obj/vehicle/tonk/Move()
+	. = ..()
+	update_icon()
+
+/obj/vehicle/tonk/update_icon() //To show damage, gun firing, whatever. We need to re apply the gun turret overlay.
+	var/icon/I = icon(icon,icon_state,dir)
+	bound_width = I.Width() //Adjust the hitbox in case admins want to make an OMEGATONK
+	bound_height = I.Height()
+	if(!turret_overlay) //We have to do this because of BYOND's innate dir setting habits, in byond, if you have an overlay'd object its direction can ONLY be which way the source is facing. So, we improvise.. ((yes this is kinda shit))
+		return
+	if(main_cannon_dir)
+		turret_overlay.setDir(main_cannon_dir)
+		if(main_cannon_dir == WEST || main_cannon_dir == SOUTHWEST|| main_cannon_dir == NORTHWEST)
+			turret_overlay.pixel_x = -19
+		else
+			turret_overlay.pixel_x = 0
+	turret_overlay.forceMove(get_turf(src))
+
+/obj/vehicle/tonk/attack_hand(mob/user)
+	if(pilot && gunner)
+		to_chat(user, "You are unable to enter [src] because all of its seats are occupied!")
+		return ..()
+	if(user in operators)
+		exit_tank(user)
+		return
+	var/position = alert("What seat would you like to enter?.",name,"pilot","gunner") //God I wish I had radials to work with
+	if(!position)
+		return
+	if(pilot && position == "pilot")
+		to_chat(user, "[pilot] is already controlling [src]!")
+		return
+	if(gunner && position == "gunner")
+		to_chat(user, "[gunner] is already controlling [src]!")
+		return
+	if(can_enter(user)) //OWO can they enter us????
+		to_chat(user, "You climb into [src] as a [position]!")
+		return enter(user, position) //Yeah i could do this with a define, but this way we're not using multiple things
+
+/obj/vehicle/tonk/proc/can_enter(var/mob/living/carbon/M) //NO BENOS ALLOWED
+	if(!M.IsAdvancedToolUser())
```

Comment:
Should probably `if(!istype(M))` to prevent silicon from using this ?

Reply:
Copied from current tankcode so iunno

Reply:
Silicons in current code are allowed to enter the tank. They can drive it, they just can't shoot due to their restrictions.
It would be necessary to check for the synth flag anyway, to prevent them, as they are carbon and human.